packages/core/src/instrumentations/base.ts

```diff
@@ -0,0 +1,14 @@
+import * as init from '../initialize';
+import type { Agent } from '../types';
+import type { Instrumentation } from './types';
+
+export abstract class BaseInstrumentation implements Instrumentation {
+  abstract initialize(): void;
+
+  get agent(): Agent {
+    return init.agent;
+  }
+
+  abstract name: string;
```

Comment:
should `name` and `version` be `static`?

Reply:
I considered it, but it makes the `Instrumentation` type ugly and forces a specific way of implementing :disappointed:  imho not worthwhile